library/psa_crypto_slot_management.c

```diff
@@ -259,15 +263,20 @@ static psa_status_t psa_load_persistent_key_into_slot( psa_key_slot_t *slot )
         data = (psa_se_key_data_storage_t *) key_data;
         memcpy( &slot->data.se.slot_number, &data->slot_number,
                 sizeof( slot->data.se.slot_number ) );
+
+        status = PSA_SUCCESS;
+        goto exit;
     }
-    else
 #endif /* MBEDTLS_PSA_CRYPTO_SE_C */
+
+    if ( key_data == NULL )
```

Comment:
I can see that this can happen but I'd say that this should rather be fixed in psa_load_persistent_key() or even psa_parse_key_data_from_storage(). psa_load_persistent_key() should return with the error PSA_ERROR_STORAGE_FAILURE if the size of key data is 0.
This is an existing bug to me and should be fixed in a separate commit.

Reply:
Done.